src/Badges.sol

```diff
@@ -121,26 +207,52 @@ contract Badges is
 
   function tokenURI(uint256 _tokenId) external view virtual override returns (string memory) {
     require(exists(_tokenId), "tokenURI: token doesn't exist");
+
     return tokenURIs[_tokenId];
   }
 
-  function unequip(uint256 _tokenId) external virtual override {
-    require(owners[_tokenId] != address(0), "unequip: token doesn't exist");
+  function unequip(uint256 _tokenId) external virtual override tokenExists(_tokenId) {
     require(msg.sender == owners[_tokenId], "unequip: sender must be owner");
+
     uint256 voucherHashId = voucherHashIds[_tokenId];
     usedHashes.unset(voucherHashId);
     burn(_tokenId);
   }
 
   function balanceOf(address _owner) external view virtual override returns (uint256) {
     require(_owner != address(0), "balanceOf: address zero is not a valid owner_");
+
     return balances[_owner];
   }
 
-  function ownerOf(uint256 _tokenId) external view virtual override returns (address) {
-    address owner_ = owners[_tokenId];
-    require(owner_ != address(0), "ownerOf: token doesn't exist");
-    return owner_;
+  function ownerOf(uint256 _tokenId) external view virtual override tokenExists(_tokenId) returns (address) {
+    return owners[_tokenId];
+  }
+
+  function revokeBadge(uint256 _raftTokenId, uint256 _badgeId) external senderIsRaftOwner(_raftTokenId) {
+    revokedBadges[_badgeId] = true;
+  }
+
+  function reinstateBadge(uint256 _raftTokenId, uint256 _badgeId)
+    external
+    tokenExists(_badgeId)
+    senderIsRaftOwner(_raftTokenId)
+  {
+    delete revokedBadges[_badgeId];
+  }
+
+  function updateExpiration(
```

Comment:
Expiration should never be updateable, no? Did you mean 

```suggestion
  function setExpiration(
```

Reply:
Also, this is not called anywhere it looks like.. we need to set this upon mint, no?

Reply:
refer to here: https://linear.app/otterspace/issue/DEV-511/dao-operator-can-revoke-deactivateexpire-a-badge

Reply:
Right, I think they are separate. Emily's comment was "we should think about what kind of process we can support to "reactivate" a badge that has expired for any reason, whether revocation or expiry on date." 

Which made me think that if a badge "naturally" expires, that there should be a way to extend the expiration date. Which is why I chose to call it "updateExpiration". And that function would only ever be called to update an expiration. 

Revoking and reinstating is separate from this.

Reply:
Ok - good. I'm aligned. 

Let's keep the concept of expire/renew separate from revoke/reinstate. Let's use this language to dintinguish them. Also, let's stick to naming with `setExpiration` instead of `updateExpiration` to keen it consistent with existing code base.